syntax = "proto3";

option go_package = "/pb";

package pb;

message BrakeManager {
    enum States {
        INIT = 0;
        IDLE = 1;
        CALIBRATION = 2;
        READY = 3;
        BRAKING = 4;
        EMERGENCY_BRAKING = 5;
    }
    States state = 1;
    uint32 hydrolic_pressure_loss = 2;
    uint32 critical_pod_acceleration_mesure_timeout = 3;
    uint32 critical_pod_deceleration_instruction_timeout = 4;
    uint32 verin_blocked = 5;
    uint32 emergency_valve_open_without_hydrolic_pressor_diminution = 6;
    uint32 critical_emergency_brakes_without_deceleration = 7;
    uint32 mesured_distance_less_than_desired = 8;
    uint32 mesured_distance_greater_as_desired = 9;
}

message MainComputer {
    enum States {
        SLEEP = 0;
        STATIC_FAULT = 1;
        DYNAMIC_FAULT = 2;
        SAFE_TO_APPROACH = 3;
        INITIALIZATION = 4;
        SAFE_TO_LAUNCH = 5;
        ACCELERATING = 6;
        EMERGENCY_BRAKING = 7;
        END_BRAKES = 8;
    }
    States state = 1;
}

message SensorState {
    MainComputer main_computer = 1;
    BrakeManager brake_manager = 2;
}